Support for HTML5 time element; option for lowercase dates

samatjain opened the pull request:
Support HTML5's time element (a preferred way to markup dates, which includes W3CTDF dates in the "datetime" attribute).

Also, add option to jQuery plugin to lowercase generated dates.

Unit tests included for both.

Sorry for adding 2 features in the same branch!

zachleat commented on src/humane.js:
```diff
@@ -103,12 +103,27 @@ function humaneDate(date, compareTo){
+            if(typeof date === 'undefined')
+                return;
+
+            date = humaneDate(date);
```
Might be better as date = humanDate(date) || '';

Then we can eliminate the typeof comparison below.

samatjain replied:
Agreed 100%.

zachleat commented on src/humane.js:
```diff
@@ -103,12 +103,27 @@ function humaneDate(date, compareTo){
-                date = humaneDate($t.attr('title'));
+                date = $t.attr('datetime') || $t.attr('title');
+
+            if(typeof date === 'undefined')
```
If(!date) should work here, no? typeof feels too restrictive. If the attributes exist but are empty strings, this will continue on its merry way.

samatjain committed:
Support for HTML5 time element

Look for HTML5 time element's "datetime" attribute first, then "title"
attribute. Includes unit test.

samatjain committed:
Lowercase output option

Add option (at jQuery plugin level) to lowercase dates (default remains
title case). Includes unit test.

samatjain commented:
Re-did both commits w/ your suggestions. Let me know if you've further suggestions.

zachleat commented:
Looks great, thanks!!
